### DISCUSSION TITLE
Update Optolink

### DISCUSSION THREAD

**Discussion by SpezialAgent** [philippoo66/optolink-splitter#57] [General]
Hallo Phil,
ich habe jetzt alle Beiträge recherchiert, habe leider aber keinen Hinweis auf ein Update gefunden. Muss ich die Installation komplett neu durchführen oder reicht es aus, einige Libraries auszutauschen?
Gruß
Martin

**Comment by philippoo66** [comment]
Hai Martin!

Das Skript wird ja garnicht installiert, von daher reicht es, geänderte (und ggf. hinzugekommene) Dateien zu übernehmen. Am einfachsten ist es, grad alle zu nehmen AUSSER der settings_ini (falls du da schon deine eigenen Einstellungen vorgenommen hast).

Aber dir geht es glaubich grad um die settings_ini wegen der Vcal Datenpunkte, richtig? da kannst du sogar auch einfach die Liste aus github kopieren und bei dir einfügen. Was anderes hat sich glaubich eh nicht geändert seit einiger Zeit...

**Reply by SpezialAgent** [reply]
Hi Phil,
es geht mir tatsächlich um die Python-Dateien/Scripte. Recht es aus, diese aus dem Repo zu installieren? Die Dateien liegen in meinem Home:

`martin@vcontrol:~/optolink-splitter-main $ ls -al
insgesamt 244
drwxr-xr-x 4 martin martin  4096 30. Nov 16:44  .
drwx------ 6 martin martin  4096 27. Nov 16:17  ..
-rw-r--r-- 1 martin martin   690 16. Nov 17:43  allupdate.sh
-rw-r--r-- 1 martin martin   189 16. Nov 17:43  .gitignore
-rw-r--r-- 1 martin martin 35149 16. Nov 17:43  LICENSE
-rw-r--r-- 1 martin martin  1003 16. Nov 17:43  list_ports.py
-rw-r--r-- 1 martin martin  4641 16. Nov 17:43  mqtt_util.py
drwxr-xr-x 5 martin martin  4096 26. Nov 16:51  myvenv
-rw-r--r-- 1 martin martin  3055 16. Nov 17:43  onewire_util.py
-rw-r--r-- 1 martin martin 11328 16. Nov 17:43  optolinkvs2.py
-rw-r--r-- 1 martin martin 11647 16. Nov 17:43  optolinkvs2_switch.py
drwxr-xr-x 2 martin martin  4096 30. Nov 16:41  __pycache__
-rw-r--r-- 1 martin martin  5616 16. Nov 17:43  README.md
-rw-r--r-- 1 martin martin  7603 16. Nov 17:43  requests_util.py
-rw-r--r-- 1 martin martin  4596 16. Nov 17:43  serlog.py
-rw-r--r-- 1 martin martin  6133 30. Nov 16:40  settings_ini_Kopie_10neu.py
-rw-r--r-- 1 martin martin  4899 27. Nov 19:28  settings_ini_Kopie_2.py
-rw-r--r-- 1 martin martin  5287 28. Nov 10:19  settings_ini_Kopie_3.py
-rw-r--r-- 1 martin martin  5586 28. Nov 17:03  settings_ini_Kopie_4.py
-rw-r--r-- 1 martin martin  5598 28. Nov 20:54  settings_ini_Kopie_5.py
-rw-r--r-- 1 martin martin  5597 29. Nov 07:28  settings_ini_Kopie_6.py
-rw-r--r-- 1 martin martin  5750 29. Nov 07:42  settings_ini_Kopie_7.py
-rw-r--r-- 1 martin martin  5717 29. Nov 15:14  settings_ini_Kopie_8.py
-rw-r--r-- 1 martin martin  6129 30. Nov 16:12  settings_ini_Kopie_9.py
-rw-r--r-- 1 martin martin  4782 27. Nov 16:42 'settings_ini Kopie.py'
-rw-r--r-- 1 martin martin  6133 30. Nov 16:40  settings_ini.py
-rw-r--r-- 1 martin martin  4736 26. Nov 17:43  settings_ini.py.old1
-rw-r--r-- 1 martin martin  4263 16. Nov 17:43  tcpip_util.py
-rw-r--r-- 1 martin martin  3753 16. Nov 17:43  utils.py
-rw-r--r-- 1 martin martin  3428 16. Nov 17:43  viconn_util.py
-rw-r--r-- 1 martin martin  4257 16. Nov 17:43  viessdata_util.py
`

`martin@vcontrol:~/optolink-splitter-main/myvenv $ ls -al
insgesamt 24
drwxr-xr-x 5 martin martin 4096 26. Nov 16:51 .
drwxr-xr-x 4 martin martin 4096 30. Nov 16:44 ..
drwxr-xr-x 3 martin martin 4096 27. Nov 13:32 bin
drwxr-xr-x 3 martin martin 4096 26. Nov 16:51 include
drwxr-xr-x 3 martin martin 4096 26. Nov 16:51 lib
lrwxrwxrwx 1 martin martin    3 26. Nov 16:51 lib64 -> lib
-rw-r--r-- 1 martin martin  180 26. Nov 16:51 pyvenv.cfg
martin@vcontrol:~/optolink-splitter-main/myvenv $ `

Die Settings-ini läuft nun sauber. Ich kann geänderte Dateien/Version per FileZilla vom Mac zum Raspi kopieren, so habe ich die Settings immer als Backup auf dem Mac.

Gruß
Martin

**Comment by SpezialAgent** [comment]
Sorry, das ist nicht lesbar. Hier als Bildschirmfotos:
<img width="586" alt="Bildschirmfoto 2024-12-05 um 21 06 00" src="https://github.com/user-attachments/assets/cdef96d5-f51e-45df-809a-7d56085fd8d9">
<img width="586" alt="Bildschirmfoto 2024-12-05 um 21 05 42" src="https://github.com/user-attachments/assets/2c64b449-2126-4aa4-9beb-0dd852b8815b">

**Reply by SpezialAgent** [reply]
Hi Phil,
ich habe die Software nach der Anleitung von Rustimation.eu installiert. Recht es aus, diese zu wiederholen und die bestehende settings wieder reinzukopieren? Ich mache mir Sorgen um die Installation der beiden Python Libraries in der virtuellen Umgebung. 
Aktuell müsste meine Installation noch aktuell sein. Ich möchte den Update Prozess nur für später dokumentieren.

Gruß
Martin

**Reply by philippoo66** [reply]
die virtuelle umgebung ist komplett im ordner myvenv. das kann alles so bleiben wie es ist. Wenn du versuchen _solltest_, sie nochmal zu installieren, wird venv meckern, dass es schon einen entsprechenden Ordner gibt. Den müsstest du erst löschen. Aber wozu - hinterher stände dann das gleiche drin wie vorher.

du kannst in deinem Ordner /optolink-splitter-main die Dateien nach Lust und Laune ersetzen, mehr brauchst du nicht machen für ein Update.

bevor du dann mit `python optolinkvs2_switch.py` den Splitter startest (sorry fpr den fürchterlich langen Dateinamen...), muss du natürlich die virtuelle Umgebung mit `source myvemv/bin/activate` starten

wenn du das ganze so installiert hast, dass es als Dienst gestartet wird, wenn der Raspi hochfährt, brauchst du für einn Update ausser dem Ersetzen der Dateien im Orner /optolink-splitter-main garnix machen.

ich hoffe ich hab deine Frage richtig verstanden und verständlich beantwortet. ansonsten musst du noch mal bescheid sagen.

Grüsse!
Phil

**Reply by SpezialAgent** [reply]
Hallo Phil,
das passt. Vielen Dank für deine Infos.
Ich hatte mit Vcontrold begonnen, die Vitocal auszulesen. Leider war das Programm fehleranfällig und stürzte spätesten nach einer Woche ab.
Der OptiLink-Splitter läuft ohne Probleme und liefert fleissig Daten.
Eine ganz herzlichen Dank für das tolle Projekt. Ich bin begeistert!!!!!
Gruß und schöne Festtage
Martin

**Reply by philippoo66** [reply]
@SpezialAgent  danke für die schöne Rückmeldung und guten Wünsche!
dir auch ne schöne Zeit & Grüsse!
Phil

ps. ich hatte es auch mal mit vcontrold versucht. aber das ist ja fürchterlich sperrig mit den ganzen xmls und erstmal compilieren und so. ich hab das nichtmal zum Laufen bekommen...
